pkg/apis/application/v1alpha1/applicationset_types.go

```diff
@@ -528,6 +536,10 @@ type SCMProviderGeneratorBitbucketServer struct {
 	AllBranches bool `json:"allBranches,omitempty" protobuf:"varint,4,opt,name=allBranches"`
 	// Credentials for AccessToken (Bearer auth)
 	BearerToken *BearerTokenBitbucket `json:"bearerToken,omitempty" protobuf:"bytes,5,opt,name=bearerToken"`
+	// Allow self-signed TLS / Certificates; default: false
+	Insecure bool `json:"insecure,omitempty" protobuf:"varint,6,opt,name=insecure"`
+	// ConfigMap key holding the trusted certificates
+	CAConfigMapKeyRef *ConfigMapKeyRef `json:"caConfigMapKeyRef,omitempty" protobuf:"bytes,7,opt,name=caConfigMapKeyRef"`
```

Comment:
Seems repetitive to have `CAConfigMapKeyRef.ConfigMapName`. I don't think it is necessary to reflect they type of the object in the property name. `CARef.ConfigMapName` aligns with existing terminology used with `SecretRef`.

Reply:
Agreed!